lib/builder/components/ShareDashboard.js

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect } from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+import {
+  toogleDashboardsMenu,
+  setAccessKey,
+  clearAccessKey,
+} from "../../actions/rootActions";
+import client from "../../../config/analysis";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+const ShareDashboards = props => {
+  useEffect(() => {
+    const getKeyFromAPI = async () => {
+      const keyName = `public-dashboard:${props.dashboardTitle}`;
+      let key = await
+        client
+          .get({
+            url: client.url('projectId', `keys?name=${keyName}`),
+            api_key: client.masterKey(),
+          });
+        
+        if (!key.length) {
+          key = await
+            client
+              .post({
+                url: client.url('projectId', 'keys'),
+                api_key: client.masterKey(),
+                params: {
+                  name: keyName,
+                  is_active: true,
+                  permitted: [ 'queries', 'saved_queries' ],
```

Comment:
those permissions are too powerful - we'll allow only saved_queries and cached_queries,
plus we'll list allowed queries by name eg. https://keen.io/projects/5011efa95f546f2ce2000000/access/keys/edit?key=D9E2872BB0841C7D080D77BA1CC6E49E07FBBF8C9312D650396711AA0B02B2F8